tools/luh2/luh2mod.py

```diff
@@ -0,0 +1,258 @@
+
+    # Drop the old boundary names to avoid confusion
+    input_dataset = input_dataset.drop(labels=['lat_bounds','lon_bounds'])
+
+    print("LUH2 dataset lat/lon boundary variables formatted and added as new variable for xESMF")
+
+    return(input_dataset)
+
+# The user will need to use a surface data set to regrid from, but the surface datasets
+# need to have their dimensions renamed to something recognizable by xESMF
+def DimensionFixSurfData(input_dataset):
+
+    # Rename the surface dataset dimensions to something recognizable by xESMF.
+    # input_dataset = surfdataset.rename_dims(dims_dict={'lsmlat':'latitude','lsmlon':'longitude'})
+    input_dataset = input_dataset.rename_dims(dims_dict={'lsmlat':'lat','lsmlon':'lon'})
+
+    # Populate the new surface dataset with the actual lat/lon values
+    # input_dataset['longitude'] = input_dataset.LONGXY.isel(latitude=0)
+    # input_dataset['latitude']  = input_dataset.LATIXY.isel(longitude=0)
+    input_dataset['lon'] = input_dataset.LONGXY.isel(lat=0)
+    input_dataset['lat'] = input_dataset.LATIXY.isel(lon=0)
+
+    print("Surface dataset dimensions renamed for xESMF")
+
+    return(input_dataset)
+
+# def SetMask(input_dataset, masking_dataset):
+
+#     # check what sort of inputdata is being provided; surface dataset or luh2
+#     # LUH2 data will need to be masked based on the variable input to mask
+#     dsflag,dstype = CheckDataset(input_dataset)
+#     if (dsflag):
+#         if(dstype == "LUH2"):
+#             SetMaskLUH2(input_dataset) # temporary
+#         elif(dstype == "Surface"):
+#             SetMaskSurfData(input_dataset)
+#         print("mask added")
+#
+#     return(input_dataset)
+
+# LUH2 specific masking sub-function
+def SetMaskLUH2(input_dataset,static_data_set):
+
+    # Mask the luh2 data where the ice/water fraction is unity (i.e. fully ice covered gridcell)
+    input_dataset["mask"] = (static_data_set.icwtr != 1)
+    # return(outputdataset)
+    return(input_dataset)
+
+# Surface dataset specific masking sub-function
+def SetMaskSurfData(input_dataset):
+    # Instead of passing the label_to_mask, loop through this for all labels?
+    input_dataset["mask"] = input_dataset["PCT_NATVEG"] > 0
+    # return(outputdataset)
+    return(input_dataset)
+
+# Check which dataset we're working with
+def CheckDataset(input_dataset):
+
+    dsflag = False
+    dsvars = list(input_dataset.variables)
+    if('primf' in dsvars or
+       'primf_to_secdn' in dsvars or
+       any('irrig' in subname for subname in dsvars)):
+        dstype = 'LUH2'
+        dsflag = True
+        print("LUH2")
+    elif('natpft' in dsvars):
+        dstype = 'Surface'
+        dsflag = True
+        print("Surface")
+    elif('icwtr' in dsvars):
+        dstype = 'static'
+        dsflag = True
+    elif('col' in dsvars):
+        dstype = 'regrid'
+        dsflag = True
+    else:
+        dstype = 'Unknown'
+        print("Unrecognize data set")
```

Comment:
@glemieux 
reminder to exit here with error message.

Reply:
resolved with 4615b8a6a16462c6712d81939a2e9e3c89eff50c